pkg/gardenlet/controller/backupentry/backupentry/reconciler.go

```diff
@@ -0,0 +1,674 @@
+		}
+		// if the extension BackupEntry doesn't exist yet, create it
+		mustReconcileExtensionBackupEntry = true
+	} else if !reflect.DeepEqual(extensionBackupEntry.Spec, extensionBackupEntrySpec) {
+		// if the extensionBackupEntry spec has changed, reconcile it
+		mustReconcileExtensionBackupEntry = true
+	} else if extensionBackupEntry.Status.LastOperation == nil {
+		// if the extension did not record a lastOperation yet, don't update the status
+		reconciliationSuccessful = false
+	} else {
+		// check for errors, and if none are present, reconciliation has succeeded
+		lastOperationState := extensionBackupEntry.Status.LastOperation.State
+		if extensionBackupEntry.Status.LastError != nil ||
+			lastOperationState == gardencorev1beta1.LastOperationStateError ||
+			lastOperationState == gardencorev1beta1.LastOperationStateFailed {
+			if lastOperationState == gardencorev1beta1.LastOperationStateFailed {
+				mustReconcileExtensionBackupEntry = true
+			}
+			reconciliationSuccessful = false
+
+			lastError := fmt.Errorf("extension state is not Succeeded but %v", lastOperationState)
+			if extensionBackupEntry.Status.LastError != nil {
+				lastError = v1beta1helper.NewErrorWithCodes(fmt.Errorf("error during reconciliation: %s", extensionBackupEntry.Status.LastError.Description), extensionBackupEntry.Status.LastError.Codes...)
+			}
+
+			lastObservedError := v1beta1helper.NewErrorWithCodes(lastError, v1beta1helper.DeprecatedDetermineErrorCodes(lastError)...)
+			reconcileErr := &gardencorev1beta1.LastError{
+				Codes:       v1beta1helper.ExtractErrorCodes(lastObservedError),
+				Description: lastObservedError.Error(),
+			}
+
+			r.Recorder.Event(backupEntry, corev1.EventTypeWarning, gardencorev1beta1.EventReconcileError, reconcileErr.Description)
+
+			description := reconcileErr.Description
+			if lastOperationState == gardencorev1beta1.LastOperationStateFailed {
+				description += ". Operation will be retried."
+			}
+			if updateErr := r.updateBackupEntryStatusError(ctx, backupEntry, operationType, description, reconcileErr); updateErr != nil {
+				return reconcile.Result{}, fmt.Errorf("could not update status after reconciliation error: %w", updateErr)
+			}
+		}
+	}
+
+	if mustReconcileExtensionBackupEntry {
+		if err := r.reconcileBackupEntryExtension(ctx, backupBucket, backupEntry, component); err != nil {
+			return reconcile.Result{}, err
+		}
+		// return early here, the BackupEntry status will be updated by the reconciliation caused by the extension BackupEntry status update.
+		return reconcile.Result{}, nil
+	}
+
+	if reconciliationSuccessful {
+		if updateErr := r.updateBackupEntryStatusSucceeded(ctx, backupEntry, operationType); updateErr != nil {
+			return reconcile.Result{}, fmt.Errorf("could not update status after reconciliation success: %w", updateErr)
+		}
+	}
+
+	if kutil.HasMetaDataAnnotation(&backupEntry.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationRestore) {
```

Comment:
Previously, this annotation was only removed if the reconciliation/restoration is successful.